Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,1195 @@
+        '''
+        if channel in synonymMap:
+            return synonymMap[channel]
+        return []
+
+    def getBSDFChannelInputName(self, bsdf, channel):
+        '''
+        Returns the name of the input for the given channel on the bsdf.
+        None if the BSDF does not support the input.
+        '''
+        if channel in bsdf.inputs:
+            return channel
+
+        for c in self.getChannelNameSynonyms(channel):
+            if c in bsdf.inputs:
+                return c
+
+        return None
+
+    def bakeChannel(self, context, obj, mat, bsdf, matOutput, channel):
+        canBakeBSDF = self.canBsdfBeBaked(bsdf)
+
+        mat = self.prepareBake(context, obj, bsdf, mat, canBakeBSDF)
+
+        # Ideally, we should bake the BSDF inputs:
+        if canBakeBSDF:
+            bakedImageNode = self.bakeChannelInputsDirectly(context, obj, mat, bsdf, matOutput, channel)
+        else:
+            bakedImageNode = self.bakeWithFallback(context, obj, mat, channel)
+
+        if not bakedImageNode:
+            return mat
+
+        # Connect baked image to baked bsdf node:
+        node_tree = mat.node_tree
+        bsdf = node_tree.nodes[mat[BAKED_MATERIAL_SHADER]]
+        inputChannelName = self.getBSDFChannelInputName(bsdf, channel)
+
+        # Connect baked image to the input now:
+        if inputChannelName == 'Normal':
+            normalMapNode = node_tree.nodes.new("ShaderNodeNormalMap")
+            normalMapNode.location = (bakedImageNode.location[0], bakedImageNode.location[1])
+            bakedImageNode.location = (bakedImageNode.location[0] - 300, bakedImageNode.location[1])
+
+            node_tree.links.new(bakedImageNode.outputs[0], normalMapNode.inputs['Color'])
+            node_tree.links.new(normalMapNode.outputs[0], bsdf.inputs[inputChannelName])
+        else:
+            node_tree.links.new(bakedImageNode.outputs[0], bsdf.inputs[inputChannelName])
+
+        return mat
+
+    def execute(self, context):
+        if not os.access(context.scene.meshsync_bake_settings.bakedTexturesPath, os.W_OK):
+            self.report({'WARNING'}, "The folder to save baked textures to does not exist!")
+            self.stop(context)
+            return {'CANCELLED'}
+
+        self.startTime = time.time()
+        self.context = context
+        self.bakeTask = self.bake()
+
+        return {'RUNNING_MODAL'}
+
+    def stop(self, context):
+        wm = context.window_manager
+        wm.event_timer_remove(self.timer)
+
+    def invoke(self, context, event):
+        wm = context.window_manager
+        self.timer = wm.event_timer_add(0, window=context.window)
+        context.window_manager.modal_handler_add(self)
+
+        # Store area for later use, these won't exist on modal timer callbacks:
+        self.area = context.area
+
+        return self.execute(context)
+
+    def modal(self, context, event):
+        # Allow cancellation by pressing escape:
+        if event.type == 'ESC':
```

Comment:
Is it documented or shown to the user that they can cancel the process by hitting ESC?

Reply:
I've put it in the docs

Reply:
Is there anywhere we can show this on the UI? Something like "In Progress (Press ESC to Cancel)"? Users will often skim docs or ignore them entirely and might miss it otherwise

Reply:
I've added it now.